app/client/components/addLecture/LectureDurationSelect.tsx

```diff
@@ -0,0 +1,67 @@
+import { zosiaApi, zosiaApiRoutes } from "@client/utils/zosiaApi";
+import { Select } from "@headlessui/react";
+import { useQuery } from "@tanstack/react-query";
+import React from "react";
+import { WidgetHandler } from "reactivated/dist/forms";
+import { DjangoFormsWidgetsSelect } from "reactivated/dist/generated";
+
+interface LectureDurationSelectProps {
+  field: WidgetHandler<DjangoFormsWidgetsSelect>;
+  lectureType: string;
+}
+
+interface LectureDurations {
+  durations: string[];
+}
+
+export const LectureDurationSelect = ({
+  field,
+  lectureType,
+}: LectureDurationSelectProps) => {
+  const { isPending, isError, data, error } = useQuery({
+    queryKey: [zosiaApiRoutes.addLectureDurations, lectureType],
+    queryFn: async ({ queryKey }) => {
+      const lecture = queryKey[1];
+      if (lecture === "") {
+        return Promise.resolve<LectureDurations>({
+          durations: ["Please choose lecture type first!"],
```

Comment:
That is especially weird because, in fact, we have an error, and outside of it, we think it is a valid response

Reply:
The desired behaviour is as follows:
- User chooses the lecture type (normal lecture or workshop)
- Based on the lecture type the user can select duration of the event (set of allowed durations is different for lectures and for workshops)

That's why if the user didn't yet choose a lecture type, `"Please choose lecture type first!"` is a valid response to a query asking "What are the allowed durations?".